docs/developer-guide/rules.md

```diff
@@ -300,6 +301,30 @@ function rule(primary, secondary) {
 }
 ```
 
+### Add `languageOptions` support
+
+Depending on the rule, it may need to support the [`languageOptions`](../user-guide/configure.md#languageoptions) configuration property.
+
+For example:
+
+```diff js
+import { basicKeywords } from '../../reference/keywords.mjs';
+
+function rule(primary, secondary) {
+  return (root, result) => {
+    /* .. */
+
+    if (!validOptions) return;
+
++   const languageCssWideKeywords = result.stylelint.config?.languageOptions?.syntax?.cssWideKeywords ?? [];
+
++   languageCssWideKeywords.forEach(basicKeywords.add, basicKeywords);
```

Comment:
Ah... wait. This code seems to mutate the shared `basicKeywords` object by a specific rule (or plugin), but is there no problem? 🤔 

```sh-session
$ node          
Welcome to Node.js v23.10.0.
Type ".help" for more information.
> let { basicKeywords } = await import("./lib/reference/keywords.mjs")
undefined
> ["foo", "bar"].forEach(basicKeywords.add, basicKeywords)
undefined
> basicKeywords
Set(7) {
  'initial',
  'inherit',
  'revert',
  'revert-layer',
  'unset',
  'foo',
  'bar'
}
```

Reply:
If we want to avoid unexpected mutations, we should do this? 🤔 

```sh-session
$ node
Welcome to Node.js v23.10.0.
Type ".help" for more information.
> let { basicKeywords } = await import("./lib/reference/keywords.mjs")
undefined
> [...basicKeywords, ...["foo", "bar"]]
[
  'initial',
  'inherit',
  'revert',
  'revert-layer',
  'unset',
  'foo',
  'bar'
]
> basicKeywords
Set(5) { 'initial', 'inherit', 'revert', 'revert-layer', 'unset' }
```

Reply:
Good catch. Let's avoid those side effects. Should we pull in the `uniteSet` util or can you suggest a better way?

Reply:
E.g.

```js
const keywords = uniteSets(
	basicKeywords,
	result.stylelint.config?.languageOptions?.syntax?.cssWideKeywords ?? [],
);
```

Or when not using `uniteSets`:

```js
const keywords = new Set([
	...basicKeywords,
	...(result.stylelint.config?.languageOptions?.syntax?.cssWideKeywords ?? []),
]);
```